If compiled as an xcframework, the package produces a non-compiling swiftinterface file

### Information

- **Package version:** 1.0.2 (or latest main)
- **Platform version:** MacOS 12.5
- **Swift version:**  swift-driver version: 1.45.2 Apple Swift version 5.6 (swiftlang-5.6.0.323.62 clang-1316.0.20.8)

### Checklist

- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] I've searched for [existing GitHub issues](https://github.com/apple/swift-collections/issues).

### Steps to Reproduce

1. Compile the package as an xcframework (with library evolution enabled). There are many ways to do this.
I used the following:

```bash
swift create-xcframework OrderedCollections --configuration debug --stack-evolution --platform ios --xc-setting CLANG_ENABLE_CODE_COVERAGE=NO --xc-setting GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO --xc-setting SWIFT_SERIALIZE_DEBUGGING_OPTIONS=NO --xc-setting OTHER_SWIFT_FLAGS="-Xfrontend -no-serialize-debugging-options" --xc-setting DEBUG_INFORMATION_FORMAT=dwarf-with-dsym
```

2. Try to use the resulting xcframework in an iOS app. Just create an empty SwiftUI based app from a template and add the line:

```swift
import OrderedCollections
```

(Please also add the xcframework to Build Phases -> Link Binary with Libraries)

3. Build (Cmd+B)

### Expected behavior

The app must compile without errors.


### Actual behavior

Compiler gives errors in the swiftinterface file of the xcframework:

```
 /Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2734:22: error: type alias 'Element' references itself
    public typealias Element = OrderedCollections.OrderedSet<Element>.Element
                     ^
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2734:62: note: while resolving type 'Element'
    public typealias Element = OrderedCollections.OrderedSet<Element>.Element
                                                             ^
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2734:71: note: while resolving type 'OrderedCollections.OrderedSet<Element>.Element'
    public typealias Element = OrderedCollections.OrderedSet<Element>.Element
                                                                      ^
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2811:1: error: type 'OrderedSet<Element>.UnorderedView' does not conform to protocol 'SetAlgebra'
extension OrderedCollections.OrderedSet.UnorderedView : Swift.SetAlgebra {
^
Swift.SetAlgebra:2:20: note: protocol requires nested type 'Element'; do you want to add it?
    associatedtype Element
                   ^
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2746:20: error: extraneous argument label '_base:' in call
      UnorderedView(_base: self)
                   ^~~~~~~~
                    
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2749:31: error: extraneous argument label '_base:' in call
      var view = UnorderedView(_base: self)
                              ^~~~~~~~
                               
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2752:14: error: cannot yield immutable value of type '_' as an inout yield of type 'OrderedSet<Element>.UnorderedView'
      yield &view
             ^
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2837:35: error: type of expression is ambiguous without more context
    let (inserted, index) = _base.append(newMember)
                            ~~~~~~^~~~~~~~~~~~~~~~~
/Users/USER/Library/Developer/Xcode/DerivedData/TestOrderedCollections-dugnwyfdwqgmnmhkxmqtpqgywhvo/Build/Products/Debug-iphonesimulator/OrderedCollections.framework/Modules/OrderedCollections.swiftmodule/x86_64-apple-ios-simulator.swiftinterface:2841:35: error: type of expression is ambiguous without more context
    let (inserted, index) = _base.append(newMember)
                            ~~~~~~^~~~~~~~~~~~~~~~~
/Users/USER/Developer/Temp/TestOrderedCollections/TestOrderedCollections/TestOrderedCollectionsApp.swift:9:8: error: failed to build module 'OrderedCollections' for importation due to the errors above; the textual interface may be broken by project issues or a compiler bug
import OrderedCollections
```

--- a/Sources/OrderedCollections/OrderedSet/OrderedSet+UnorderedView.swift
+++ b/Sources/OrderedCollections/OrderedSet/OrderedSet+UnorderedView.swift
@@ -14,8 +14,6 @@ extension OrderedSet {
   /// conformance.
   @frozen
   public struct UnorderedView {
-    public typealias Element = OrderedSet.Element
-
     @usableFromInline
     internal var _base: OrderedSet
 
@@ -138,7 +136,9 @@ extension OrderedSet.UnorderedView: ExpressibleByArrayLiteral {
   }
 }
 
-extension OrderedSet.UnorderedView: SetAlgebra {}
+extension OrderedSet.UnorderedView: SetAlgebra {
+  public typealias Element = OrderedSet.Element
+}
 
 extension OrderedSet.UnorderedView {
   /// Creates an empty set.
@@ -302,7 +302,7 @@ extension OrderedSet.UnorderedView {
   @inlinable
   @inline(__always)
   @discardableResult
-  public mutating func remove(_ member: Self.Element) -> Self.Element? {
+  public mutating func remove(_ member: Element) -> Element? {
     _base.remove(member)
   }
 }

